text/0000-multiarch-builders-and-package.md

```diff
@@ -0,0 +1,944 @@
+exported and combined into an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Buildpack Package
+
+As a quick summary, our current process to create a buildpack package involves:
+
+- The end-users defined `os` for the OCI image using the [package.toml](https://buildpacks.io/docs/reference/config/package-config/) when they want to export a buildpack package.
+- The only values allowed are `linux` and `windows` and by default when is not present, `linux` is being used.
+- When exporting to daemon, the `docker.OSType` must be equal to `platform.os`
+
+### To keep compatibility 
+
+We propose:
+- Deprecate the `platform.os` field from [package.toml](https://buildpacks.io/docs/reference/config/package-config/). It will be removed after two pack releases with the new feature
+- When `platform.os` is present in [package.toml](https://buildpacks.io/docs/reference/config/package-config/), throw a warning messages indicating the field will be removed 
+and `--platform` flag must be used
+- When `platform.os` is not present [package.toml](https://buildpacks.io/docs/reference/config/package-config/) and `--platform` flag is not used, throw a warning messages indicating 
+a new `--platform` flag is available
+- Keep doing our current process to package a buildpack
+
+### To improve user experience
+
+We propose:
+- Add a new `--platform` flag with format `[os][/arch][/variant]:[name@version]` to build for a particular target, once the `platform.os` field is removed, 
+this will be the way for end-users to specify the platform for which they want to create single OCI artifact.
+> **Note**
+> We want to avoid duplicated information, we expect to have `targets` with all the platform supported by a buildpack
+
+- Add a new boolean `--multi-arch` flag to indicate pack it must create multiples OCI artifacts and combine them with an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md), 
+this flag must be used in conjunction with `--publish` or `--format file` flags and error out when `daemon` is selected.
+> **Note**
+> I got feedback to avoid creating an Image Index if users are not expecting that, the flag is an acknowledgment for creating the index 
+
+- A new folder structure to organize the buildpacks binaries for multi-arch images
+```bash
+.
+├── buildpack.toml
+└── {os}-{arch}             // optional
+    └── {variant}           // optional
+        ├── {version-1}     // optional
+        │   └── bin
+        │       ├── build
+        │       └── detect
+        └── {version-2}     // optional
+            └── bin
+                ├── build
+                └── detect
+```
+> **Note** 
+> For cross-compile buildpacks like Paketo, it looks easy to add a step to their Makefile to compile and separate the binaries following this structure.
```

Comment:
Yes, that shouldn't be an issue.

Would this only change the buildpacks structure presented to `pack` or would it also change the buildpack structure in the actual image that's generated? The latter would require buildpack API/spec changes, right?

Reply:
This will only change the buildpack structure presented to `pack`, the final image should keep the structure as we defined in the spec.